receiver/awscontainerinsightreceiver/internal/k8sapiserver/utils.go

```diff
@@ -127,3 +127,17 @@ func parseDeploymentFromReplicaSet(name string) string {
 	}
 	return name[:lastDash]
 }
+
+func isHyperPodNode(instanceType string) bool {
+	return strings.HasPrefix(instanceType, "ml.")
```

Comment:
We were using the nodeName at the beginning but that made the integration tests no feasible. that's why we changed to instance type instead

Reply:
There are other labels we could capture to check, but again storing additional labels for all nodes is a burden on memory, and due to the Integ test we can't use NodeNames since EKS does not support `--hostname-override`  to be anything other than the InstanceId, thus we cannot use it for a mocked hyperpod cluster.